### PR TITLE
[cypress] Use GitHub action service for Nextcloud server

### PR DESCRIPTION
**Pull request by susnux**
## Summary

Follow up on https://github.com/nextcloud/server/pull/41250 

This changes the Cypress workflow to not start a docker container from Cypress but use a GitHub workflow service as the Nextcloud container.
This saves ~2 minutes CI time per Cypress runner (currently 3).

Also the service is up when the workflow starts meaning cypress does not have to wait for it and should no longer time out.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -84,6 +84,13 @@ jobs:
 
     name: runner ${{ matrix.containers }}
 
+    services:
+      nextcloud:
+        image: ghcr.io/nextcloud/continuous-integration-shallow-server
+        options: --name nextcloud
+        ports:
+          - 80:80
```
Comment (skjnldsv) [inline .github/workflows/cypress.yml:92]: That means component will also have a server started

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -104,11 +111,10 @@ jobs:
         uses: cypress-io/github-action@59810ebfa5a5ac6fcfdcfdf036d1cd4d083a88f2 # v6.5.0
         with:
           component: ${{ matrix.containers == 'component' }}
-          group: ${{ matrix.use-cypress-cloud && matrix.containers == 'component' && 'Run component' || matrix.use-cypress-cloud && 'Run E2E' || '' }}
-          # cypress env
-          ci-build-id: ${{ matrix.use-cypress-cloud && format('{0}-{1}', github.sha, github.run_number) || '' }}
-          tag: ${{ matrix.use-cypress-cloud && github.event_name || '' }}
```
Comment (skjnldsv) [inline .github/workflows/cypress.yml:110]: This is not related

**Comment by skjnldsv** [comment]
> This saves ~2 minutes CI time per Cypress runner (currently 3).

Do you know why? I mean, both are running docker? This look like a crazy difference :exploding_head:

File: .github/workflows/cypress.yml (hunk 2 above)
Comment (susnux) [reply .github/workflows/cypress.yml:110]: But invalid as there is no `matrix.use-cypress-cloud`

**Comment by susnux** [comment]
> Do you know why? I mean, both are running docker? This look like a crazy difference 🤯

I guess because services are cached and created on the beginn while manually starting docker happens on step evaluation

File: .github/workflows/cypress.yml (hunk 1 above)
Comment (susnux) [reply .github/workflows/cypress.yml:92]: Yes, sadly github does not allow conditional services

File: .github/workflows/cypress.yml (hunk 2 above)
Comment (skjnldsv) [reply .github/workflows/cypress.yml:110]: Right, but the group, ci build and tag are still mandatory and relevant to have

Comment (susnux) [reply .github/workflows/cypress.yml:110]: Well as the variable does not exist all of them are set to ''
So it makes no difference

**Review by max-nextcloud — APPROVED** [review]
Interesting approach.

I'd be happy to save 2 min per cypress runner.

I'm wondering if there are any downsides to this but can't think of any. The only thing that comes to my mind is that we may need to maintain multiple images at some point so once this lands on stable branches they still have a container that works for them. But i guess we'll need the images anyway and can just configure them in the cypress.yml accordingly.

File: .github/workflows/cypress.yml (hunk 2 above)
Comment (skjnldsv) [reply .github/workflows/cypress.yml:110]: Doesn't mean we should dro pit instead of fixing it :thinking: 

- https://github.com/nextcloud/server/pull/40736 changed the behaviour
- https://github.com/nextcloud/server/pull/41039 changed it too but broke the group/tag 

I'd suggest we revert back to

```suggestion
          group: Run ${{ matrix.containers == 'component' && 'component' || 'E2E' }}
          # cypress env
          ci-build-id: ${{ github.sha }}-${{ github.run_number }}
          tag: ${{ github.event_name }}
```

Comment (susnux) [reply .github/workflows/cypress.yml:110]: makes sense

**Review by skjnldsv — APPROVED** [review]

File: .github/workflows/cypress.yml (hunk 2 above)
Comment (susnux) [reply .github/workflows/cypress.yml:110]: Well this breaks the CI, as those flags are only allowed when using the cypress cloud which we do not.

ref: https://docs.cypress.io/guides/references/error-messages#The---ci-build-id---group---tag---parallel-or---auto-cancel-after-failures-flags-can-only-be-used-when-recording

Comment (skjnldsv) [reply .github/workflows/cypress.yml:110]: Time for our custom cloud :cry:

File: .github/workflows/cypress.yml [hunk 3]
```diff
@@ -103,12 +110,14 @@ jobs:
       - name: Run ${{ matrix.containers == 'component' && 'component' || 'E2E' }} cypress tests
         uses: cypress-io/github-action@ebe8b24c4428922d0f793a5c4c96853a633180e3 # v6.6.0
         with:
+          ci-build-id: ${{ github.sha }}-${{ github.run_number }}
```
Comment (skjnldsv) [inline .github/workflows/cypress.yml:113]:
```suggestion
          # Unavailable without cypress cloud
          # ci-build-id: ${{ github.sha }}-${{ github.run_number }}
```

File: .github/workflows/cypress.yml [hunk 4]
```diff
@@ -103,12 +110,14 @@ jobs:
...
           component: ${{ matrix.containers == 'component' }}
-          group: ${{ matrix.use-cypress-cloud && matrix.containers == 'component' && 'Run component' || matrix.use-cypress-cloud && 'Run E2E' || '' }}
-          # cypress env
-          ci-build-id: ${{ matrix.use-cypress-cloud && format('{0}-{1}', github.sha, github.run_number) || '' }}
-          tag: ${{ matrix.use-cypress-cloud && github.event_name || '' }}
+          group: Run ${{ matrix.containers == 'component' && 'component' || 'E2E' }}
+          tag: ${{ github.event_name }}
```
Comment (skjnldsv) [inline .github/workflows/cypress.yml:116]:
```suggestion
          # Unavailable without cypress cloud
          # group: Run ${{ matrix.containers == 'component' && 'component' || 'E2E' }}
          # tag: ${{ github.event_name }}
```

File: .github/workflows/cypress.yml (hunk 2 above)
Comment (skjnldsv) [reply .github/workflows/cypress.yml:110]: https://github.com/nextcloud/server/commits/39158c0a52c478df989c955603c643f3905a7343

**Comment by skjnldsv** [comment]
Seems like it's failing much more randomly with that PR  :thinking:

**Comment by skjnldsv** [comment]
`navigation-bar-settings.cy.ts` is really failing

**Comment by skjnldsv** [comment]
`Error response from daemon: No such container: nextcloud-cypress-tests-server`
